src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpInitializeStaticFieldsInline.cs

```diff
@@ -17,7 +17,10 @@ public sealed class CSharpInitializeStaticFieldsInlineAnalyzer : InitializeStati
         protected override bool InitialiesStaticField(SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
         {
             var assignmentNode = (AssignmentExpressionSyntax)node;
-            return semanticModel.GetSymbolInfo(assignmentNode.Left, cancellationToken).Symbol is IFieldSymbol leftSymbol && leftSymbol.IsStatic;
+
+            return assignmentNode.FirstAncestorOrSelf<AssignmentExpressionSyntax>(x => x.IsKind(SyntaxKind.AddAssignmentExpression)) == null &&
```

Comment:
Handles only add assignment because I couldn't find any valid use-case where we would like to unsubscribe from the static ctor.

Reply:
Why not generalize it to any lambda?

Reply:
I was thinking about that but I thought that would be too wide exclusion. Most of the lambdas could be put in function removing this issue. But I'm happy to change the implem.

Reply:
Do you have an example of such a lambda? It seems like CA1810 should be handling the "easy" cases, while more complicated cases (enforcing the lack of `beforefieldinit`) should be covered by PerformanceSensitiveAnalyzers.

Reply:
I was thinking about some patterns I have seen where people did initialization of `Func` and `Action` in the static ctor. I don't really think that matters much especially in regard to the second part of your comment.

Reply:
Ah, I wasn't referring to the initialization of a field with a delegate type. I was referring to a lambda in the static constructor which contains a nested statement that assigns a value to a field. That assignment cannot trivially be moved out of the lambda to become an inline initialization.

Reply:
Example:

```
class C {
  static int s;
  static C() {
    Task.Run(() => s = 3);
  }
}
```

Reply:
Makes sense, I'll update the PR